yapapi/services/service_runner.py

```diff
@@ -189,6 +189,14 @@ def _change_state(
 
         if isinstance(event, tuple):
             instance.service._exc_info = event
+
+        if instance.state != prev_state:
+            ctx = instance.service._ctx
+            assert ctx is not None  # This is None only while pending
```

Comment:
```suggestion
            assert ctx is not None  # This is None only while pending for the first time
```

the instance could be pending after having been restarted from some other state